### ISSUE TITLE
Block Widgets: Cookie & Consents Banner Widget - Widget is not visible in customizer on AT sites

### ISSUE THREAD

**Issue by yashitamittal11** [Automattic/wp-calypso#57415]
### Quick summary

When adding the Cookies and Consent Banner to the widgets section, you will see that the widget is not visible in the customizer once the settings are saved.

### Steps to reproduce

1. On a WordPress.com site on the business plan, go to Appearance > Customize > Widgets > [Widget Area] > + Add a Widget
2. Add the Cookies and Consents Banner block widget.
3. Widget is not visible after saving the changes and the cookie banner close button on the frontend does not work.

### What you expected to happen

I expect the widget to show in the customizer after saving the changes. And, the banner to close on clicking the close button in the frontend.

### What actually happened

Widget is not visible in the widget section in the customizer although the cookie banner shows on the frontend. The banner does not close on the user's site on clicking the Close button.

### Context

User reported the issue in 4408091-zd-woothemes
I am able to consistently reproduce the issue on AT sites. I tested with Twenty Twenty One, Varia themes.

https://user-images.githubusercontent.com/21078987/139184605-78ec8691-2ae1-4db7-b7f4-0fdf58316e4f.mov

Banner does not close-

https://user-images.githubusercontent.com/21078987/139186139-b1b05d4b-f986-4496-b23c-e0618721c4fb.mov

I am not able to reproduce the banner close issue on my test site. I did the plugin conflict test on the user's site via the Health Check plugin but I still see the banner issue.

### Simple, Atomic or both?

Atomic

### Theme-specific issue?

No

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by kimerlin81** [comment]
Referenced Slack: p1527522152000210-slack-C03TY6J1A
TLDR: Having more than one cookie banner widget will cause unexpected results

- User did indeed have 2 cookie consent widgets added to the Footer One widget section in the Customizer.
Screen Recording: 
https://user-images.githubusercontent.com/83732167/139231542-36048d03-bec4-4ed5-9c3b-0573656d433e.mp4

- Removed one of the banner widgets and now the banner closes on the site
Screen Recording: 
https://user-images.githubusercontent.com/83732167/139231704-ec198e1f-d1c9-408e-b92b-4bb84d41a90f.mp4

- As a test, I installed and activated the Classic Widget plugin and was able to then see the legacy Cookie & Consent banner widget settings in the Customizer.
However, new block widgets like the Paragraph block looked weird as they were converted to HTML code.
Screen Recording:
https://user-images.githubusercontent.com/83732167/139232370-a2d25f35-9acb-4324-8a0d-91b2581a0113.mp4

- uninstalled Classic Widget plugin

**I believe this ticket can still stay open since this legacy widget isn't visible unless clicked in the Customizer.**
Ticket updated: 4408091-zd-woothemes

**Comment by kimerlin81** [comment]
This is related to https://github.com/Automattic/wp-calypso/issues/56468

**Comment by daria2303** [comment]
Antoher one : 4444567-zd-woothemes 
user repeated widgets because didn't see them in the Customizer, it is a bit easier to edit them in wp-admin view

**Comment by rachelwinspear** [comment]
Another report: #32722316-hc

**Comment by daria2303** [comment]
One more, text widget, simple site #32946455-hc
https://d.pr/v/bWWsf1
Any update on this?

**Comment by arcangelini** [comment]
It does look like there are some inconsistencies around how the preview is showing in the Legacy block

| Jetpack | Simple site | Atomic |
|---|---|---|
| <img width="1391" alt="Markup 2021-12-03 at 13 32 33" src="https://user-images.githubusercontent.com/33258733/144603470-58bc13ff-4184-4602-b78f-b2ac82fc7da5.png"> | <img width="1105" alt="Markup 2021-12-03 at 13 33 47" src="https://user-images.githubusercontent.com/33258733/144603539-abf69f86-0400-4534-afd3-fd9fe8de9c1d.png"> | <img width="788" alt="Markup 2021-12-03 at 13 33 10" src="https://user-images.githubusercontent.com/33258733/144603617-4a2bf24e-70b7-4957-ad1c-1d5f5d6133c9.png"> | 

Obviously, the AT outcome appears to be broken but I am not sure which is more desirable Jetpack or Simple version. Maybe @jeherve could weigh in on his perspective for Jetpack. Then I will work to either make everything like Jetpack or everything like Simple. IMO I think the preview being displayed in Jetpack is not necessary and looks broken. I can't imagine a scenario where it would be desired.

>One more, text widget, simple site #32946455-hc

@daria2303 this may be a little different situation. The text widget is disabled completely from being added to the Legacy block, the proper solution here would be to get them to use the transform option to switch that Legacy Widget block to a Paragraph and Heading blocks. They should be able to click `Convert to blocks` to make that happen.

<img width="379" alt="Markup 2021-12-03 at 13 55 48" src="https://user-images.githubusercontent.com/33258733/144606174-0eed484c-a491-4cde-a72b-2d2f11073370.png">

**Comment by jeherve** [comment]
> Obviously, the AT outcome appears to be broken but I am not sure which is more desirable Jetpack or Simple version. Maybe @jeherve could weigh in on his perspective for Jetpack.

I'd say they're all a bit broken :) But maybe the Jetpack experience is a bit less broken at the moment?

**Comment by rachelwinspear** [comment]
Another report #33161828-hc

**Comment by arcangelini** [comment]
OK the source of the issue appears to be the `setHeight` function for the preview iframe [which is in Gutenberg](https://github.com/WordPress/gutenberg/blob/trunk/packages/widgets/src/blocks/legacy-widget/edit/preview.js#L61). The preview is getting the height of the body of the iframe that is loaded with the widget instance. Since the cookie consent widget is not static on the footer it leaves the iframe with a body that is empty, making the `setHeight` function return `0`.

I had some success with stopping the normal widget load if it detects the widget is being loaded in the block and instead just echoing out the text inside a different div. This is the result in Jetpack and Atomic. What do you think @jeherve ? I can make the necessary PR's if you think this is a good start.

### Customizer
<img width="1366" alt="Markup 2021-12-14 at 09 30 15" src="https://user-images.githubusercontent.com/33258733/146029130-290dbbc8-24b3-4ee7-a55a-fb68a862e377.png">

### Widgets.php
<img width="1365" alt="Markup 2021-12-14 at 09 30 42" src="https://user-images.githubusercontent.com/33258733/146029336-8d4897a8-26e2-42ab-8997-7eaa1179fa61.png">

**Comment by jeherve** [comment]
This doesn't look bad, nice work 👍

**Comment by arcangelini** [comment]
Change ready in Jetpack waiting on review Automattic/jetpack#22094
Also ready on WPCOM for simple sites ready to review D72102-code

I will close this when both are deployed : )
